ingress/controllers/nginx-third-party/nginx/utils.go

```diff
@@ -116,3 +119,63 @@ func toMap(iface interface{}) (map[string]interface{}, bool) {
 
 	return map[string]interface{}{}, false
 }
+
+func checkChanges(filename string, data *bytes.Buffer) (bool, error) {
```

Comment:
it's kind of add that this function takes a file path and a buffer, only to convert the filepath into a buffer. Maybe just make it a method on the nginx manager called needsReload so it has access to self.config?